src/quo2/foundations/colors.cljs

```diff
@@ -260,7 +257,34 @@
   [s]
   (and (string? s) (string/starts-with? s "#")))
 
-(def custom-color
+(def resolve-color
+  "(resolve-color color theme opacity)
+   color   hex string or keyword (resolves from custom, network and semantic colors)
+   theme  :light/:dark
+   opacity 0-100 (optional) - if set theme is ignored and goes to 50 suffix internally"
+  (memoize
+   (fn
+     ([color]
+      (resolve-color color nil nil))
+     ([color theme-or-opacity]
+      (if (keyword? theme-or-opacity)
+        (resolve-color color theme-or-opacity nil)
+        (resolve-color color nil theme-or-opacity)))
+     ([color theme opacity]
+      (let [suffix         (cond opacity         50
+                                 (= :dark theme) 60
+                                 :else           50)
+            hex?           (not (keyword? color))
```

Comment:
I don't know about that, we should not be passing a string unless it has # at the start, perhaps we can add some validation on it :)

Reply:
I agree we shouldn't be passing a valid color, but that's what the code reads in the PR. It says `hex?`, but the check is only verifying it's not a keyword. In case we want to be a bit more precise in the validation, we have the function `hex-string?`. I guess we could use it, right?

Reply:
Yeah definitely, I will add that!

Reply:
hmm, I need to double check about adding this. I will say a function like this will benefit a lot from Mali 👍